libs/errno.go

```diff
@@ -9,185 +12,212 @@ import (
-#define ENOENT 46
-// Executable file format error.
-#define ENOEXEC 47
-// No locks available.
-#define ENOLCK 48
-// Reserved.
-#define ENOLINK 49
-// Not enough space.
-#define ENOMEM 50
-// No message of the desired type.
-#define ENOMSG 51
-// Protocol not available.
-#define ENOPROTOOPT 52
-// No space left on device.
-#define ENOSPC 53
-// No STREAM resources.
-#define ENOSR 54
-// Not a STREAM.
-#define ENOSTR 55
-// Functionality not supported.
-#define ENOSYS 56
-// The socket is not connected.
-#define ENOTCONN 57
-// Not a directory or a symbolic link to a directory.
-#define ENOTDIR 58
-// Directory not empty.
-#define ENOTEMPTY 59
-// Env not recoverable.
-#define ENOTRECOVERABLE 60
-// Not a socket.
-#define ENOTSOCK 61
-// Not supported (may be the same value as EOPNOTSUPP).
-#define ENOTSUP 62
-// Inappropriate I/O control operation.
-#define ENOTTY 63
-// No such device or address.
-#define ENXIO 64
-// Operation not supported on socket (may be the same value as ENOTSUP).
-#define EOPNOTSUPP 65
-// Value too large to be stored in data type.
-#define EOVERFLOW 66
-// Previous owner died.
-#define EOWNERDEAD 67
-// Operation not permitted.
-#define EPERM 68
-// Broken pipe.
-#define EPIPE 69
-// Protocol error.
-#define EPROTO 79
-// Protocol not supported.
-#define EPROTONOSUPPORT 80
-// Protocol wrong type for socket.
-#define EPROTOTYPE 81
-// Result too large.
-#define ERANGE 82
-// Read-only file system.
-#define EROFS 83
-// Invalid seek.
-#define ESPIPE 84
-// No such process.
-#define ESRCH 85
-// Reserved.
-#define ESTALE 86
-// Stream ioctl() timeout.
-#define ETIME 87
-// Connection timed out.
-#define ETIMEDOUT 88
-// Text file busy.
-#define ETXTBSY 89
-// Operation would block (may be the same value as EAGAIN).
-#define EWOULDBLOCK 90
-// Cross-device link. 
-#define EXDEV 91
-`,
+			Header: errnoHeader(),
 			Imports: map[string]string{
 				"libc": RuntimeLibc,
 			},
-			Idents: map[string]*types.Ident{
-				"errno":    c.NewIdent("errno", "libc.Errno", libc.Errno, gint),
-				"strerror": c.NewIdent("strerror", "libc.StrError", libc.StrError, c.FuncTT(c.C().String(), gint)),
-			},
+			Idents: errnoIdents(c),
 		}
 	})
 }
+
+type errnoInf struct {
+	name  string
+	value int
+}
+
+func errnoHeader() string {
+
+	lines := []string{
+		"#include <" + BuiltinH + ">",
+		"_cxgo_go_int errno = 0;",
+		"char* strerror (_cxgo_go_int errnum);",
+	}
+
+	for _, v := range errnoInfs {
+		lines = append(lines, fmt.Sprintf("const int %s = %d;", v.name, v.value))
+	}
+	res := strings.Join(lines, "\n")
+	return res
+}
+
+func errnoIdents(c *Env) map[string]*types.Ident {
+	gint := c.Go().Int()
+
+	res := map[string]*types.Ident{
+		"errno":    c.NewIdent("errno", "libc.Errno", libc.Errno, gint),
+		"strerror": c.NewIdent("strerror", "libc.StrError", libc.StrError, c.FuncTT(c.C().String(), gint)),
+	}
+
+	for _, v := range errnoInfs {
+		res[v.name] = c.NewIdent(v.name, "libc."+v.name, libc.ERRNOCONST, gint)
```

Comment:
```suggestion
		res[v.name] = c.NewIdent(v.name, "libc."+v.name, libc.Errno, gint)
```

It's okay to use `libc.Errno` here and remove `ERRNOCONST` from libc package.